File: core/src/main/java/com/illuzionzstudios/customfishing/reward/loader/FishingItemLoader.java
package com.illuzionzstudios.customfishing.reward.loader;

import com.illuzionzstudios.customfishing.reward.FishingItem;
import com.illuzionzstudios.mist.config.ConfigSection;
import com.illuzionzstudios.mist.item.loader.BaseCustomItemLoader;

/**
 * Load a fishing item from a config section
 */
public class FishingItemLoader extends BaseCustomItemLoader<FishingItem> {

    public FishingItemLoader(ConfigSection section) {
        super(section);
    }

    @Override
    protected FishingItem returnImplementedObject(ConfigSection configSection) {
        // Lets try build the reward
        FishingItem item = new FishingItem();
        item.setChance(section.getDouble("chance"));

        return item;
    }
}

File: src/main/kotlin/com/illuzionzstudios/customfishing/reward/fishing/loader/FishingItemLoader.kt
package com.illuzionzstudios.customfishing.reward.fishing.loader

import com.illuzionzstudios.customfishing.reward.fishing.item.FishingItem
import com.illuzionzstudios.mist.config.ConfigSection
import com.illuzionzstudios.mist.item.loader.BaseCustomItemLoader

class FishingItemLoader(section: ConfigSection) : BaseCustomItemLoader<FishingItem>(section) {

    override fun returnImplementedObject(section: ConfigSection?): FishingItem {
        // Lets try build the reward
        val item = FishingItem()
        item.chance = section?.getDouble("chance") ?: 0.0

        return item
    }
}